tests/virtualization/livemigration/live_migration_src.pm

```diff
@@ -0,0 +1,279 @@
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for source host
+# The live_migration_src.pm module works together with prepare_profile, boot_from_pxe, installation and login_console
+# to fulfill live migration test. The latter 4 modules are meant for src/dst host insatllation and configurationsrc.
+#
+# live_migration_src.pm features:
+# - Create VMs from config xml if reusable or insatll via autoyast which is created from template.
+# - Check if VMs are created or insatlled, ready for migration test
+# - Set up passwordless ssh authentication login dst host
+# - Execute live migration test (xen/kvm)
+# - clean up
+#
+# note:
+#  1. A external nfs shared storage can be specified by NFS_PATH. If not specified, /tmp/vm_images on src host will be exported.
+#  2. VMs are installed concurrently. Migration test and VM insatllation occur concurrently too.
+#  3. This test job can be scheduled without hosts installation by appending "IPMI_DO_NOT_RESTART_HOST=1 HOST_INSTALL=0".
+#  4. If CREATE_PARTITIONS=1 is specified /tmp/vm_images on hosts will be deleted before host installation.
+#  5. Vars like: REPO_SLES15SP3FV, REPO_SLES15SP3KVM can be specified to add repos into respective VMs during installation.
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_src;
+use strict;
+use warnings;
+use base 'opensusebasetest';
+use testapi;
+use version_utils 'is_sle';
+use Utils::Systemd 'systemctl';
+use utils qw(zypper_call zypper_ar);
+use File::Copy 'copy';
+use File::Path 'make_path';
+use lockapi;
+use mmapi;
+use scheduler 'get_test_suite_data';
+use virt_autotest::utils 'collect_virt_system_logs';
+
+my $vms = {};    #the totel number of vm to be tested.
+my $images_path = "/var/lib/libvirt/images";
+my $sut_ip = get_var('SUT_IP');
+my $inst_src = get_test_suite_data()->{location};
+my $dst_ip;
+
+#Generate a autoyast profile for a vm to be insatlled.
+sub create_profile {
+    my ($guest, $vm_name, $version, $arch, $vdisk_type, $ltss_code) = @_;
+    #    my ($guest, $vm_name, $version, $arch, $vdisk_type) = @_;
+    my $nfs_path = get_var("NFS_PATH") ? get_var("NFS_PATH") : get_var('SUT_IP') . ":/tmp/vm_images";
+    my $path = $version >= 15 ? "virtualization/livemigration/guest_15.xml.ep" : "virtualization/livemigration/guest_12.xml.ep";
+    my $scc_code = get_required_var("SCC_REGCODE");
+    my $profile = get_test_data($path);
+    $profile =~ s/\{\{SCC_REGCODE\}\}/$scc_code/g;
+    $profile =~ s/\{\{ARCH\}\}/$arch/g;
+    $profile =~ s/\{\{VERSION\}\}/$version/g;
+    $profile =~ s/\{\{SUT_IP\}\}/$sut_ip/g;
+    $profile =~ s/\{\{GUEST\}\}/$vm_name/g;
+    $profile =~ s/\{\{DISK_TYPE\}\}/$vdisk_type/g;
+    $profile =~ s/\{\{IP_PATH\}\}/$nfs_path\/guestip/g;
+    my $vars = {
+        vm_name => $vm_name,
+        ltss_code => $ltss_code,
+        repos => [split(/,/, get_var("REPO_" . uc($vm_name =~ s/-//gr), ''))]
+    };
```

Comment:
My comment in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13071/files#r740676604 is related to this create_profile function.